promoter/promoter.go

```diff
@@ -85,3 +86,42 @@ func (p *Promoter) initBackgroundThreads(f updateFunc) {
 		p.threadedAddressWatcher(p.bgCtx, f)
 	}()
 }
+
+// staticAddrDiff returns a diff of addresses that describes which addresses
+// need to be added and removed from skyd to match the state of the database.
```

Comment:
Follow Up: might be worth adding some additional context here on the design. Initially I would have thought that skyd is the source of truth for address, and so it is the db that needs to be updated. 

Is there a situation where that diff is needed? And so we should have two separate diff functions depending on the diff needed?

Reply:
Added another sentence to make it clear that we expect skyd to match the state of the database.

We never need to update the database according to the state in skyd. Only the other way round.